GrowthBook/src/jvmTest/kotlin/com/sdk/growthbook/tests/GBNumberTests.kt

```diff
@@ -0,0 +1,50 @@
+package com.sdk.growthbook.tests
+
+import com.sdk.growthbook.model.GBArray
+import com.sdk.growthbook.model.GBNumber
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class GBNumberTests {
+
+    private val integerNumbers: List<GBNumber> =
+        listOf(123, 123L, 123.toByte(), 123.toShort()).map(::GBNumber)
+
+    private val floatNumbers: List<GBNumber> = listOf(123f, 123.0).map(::GBNumber)
+
+    private val floatingPointErrorNumbers: List<GBNumber> =
+        listOf(0.1 + 0.2, 0.3, 0.1f + 0.2f).map(::GBNumber)
+
+    @Test
+    fun testEquals() {
+        assertAllPairsEqual(integerNumbers)
+        assertAllPairsEqual(floatNumbers)
+        assertAllPairsEqual(floatingPointErrorNumbers)
+    }
+
+    @Test
+    fun testHashCode() {
+        assertAllPairsEqual(integerNumbers.map { it.hashCode() })
+        assertAllPairsEqual(floatNumbers.map { it.hashCode() })
+        assertAllPairsEqual(floatingPointErrorNumbers.map { it.hashCode() })
+    }
```

Comment:
Here (and in other places in these tests) it's possible to get rid of duplication.